### PR TITLE
Bump expensify-common to 2.0.101

### PR DESCRIPTION
**Pull request by NJ-2020**
### Details
In this [PR](https://github.com/Expensify/expensify-common/pull/804#issuecomment-2377539958) we decided to match a room name if they are preceded by newline/space.

### Fixed Issues
$ https://github.com/Expensify/App/issues/46299
PROPOSAL: https://github.com/Expensify/App/issues/46299#issuecomment-2311775536

### Tests
1. Open Expensify App
2. Navigate to any chat e.g. a workspace chat.
3. Enter the message `#4856₹&_#7553` in composer and verify that only `#4856` is highlighted.
4. Send the message and verify that the Concierge message mentions only `#4856` for room creation.
5. Repeat Steps (2) and (3) and verify the expected result:
   a. Message: `#gh4h_+&&&-#ehhh`, Expected Result: Only `#gh4h`
   b. Message: `#4856₹& #7553`, Expected Result: `#4856`  and `#7553`
   c. Message: `#4856₹&`
   `#7553`, Expected Result: `#4856`  and `#7553`
   d. Message: `#4856₹&*#7553`, Expected Result: Only `#4856`
   e. Message: `#4856₹&~#7553`, Expected Result: Only `#4856`
   f. Message: `#4856₹& *#7553`, Expected Result: `#4856`  and `#7553`
   g. Message: `#4856₹& ~#7553`, Expected Result: `#4856`  and `#7553`
   h. Message: `#4856₹&`
   `*#7553`, Expected Result: `#4856`  and `#7553`
   i. Message: `#4856₹&`
   `~#7553`, Expected Result: `#4856`  and `#7553`
   j. Message: `#4856₹&`
   `_#7553`, Expected Result: `#4856`  and `#7553`
   k. Message: `#4856₹&`
   `-#7553`, Expected Result: Only `#4856`
   l. Message: `#room-name123`, Expected Result: `#room-name123`
   m. Message: `https://abc.com/abc#qwer`, Expected Result: Nothing is highlighted

- [X] Verify that no errors appear in the JS console

### Offline tests
Same as tests

### QA Steps
Same as tests

- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/69086782-dfd8-47da-826e-3b157e9e76d3

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/46dbcd09-c8de-410c-a0ed-ac6a9bbf49cd

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/ea3f7a35-6526-4297-9981-755b7c43a14c

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/bb68f9b2-68ad-49b2-b274-dc264af07488

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/2880c0c4-80d9-4fbc-887e-c55b6fdff325

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/5c10a299-39b7-4d2a-820a-0bde2af3d467

</details>

### CODE REVIEW

**Review by rojiphil — COMMENTED** [review]
@NJ-2020 The version bump from `0.1.164` to `0.1.173` for the live-markdown library would also introduce other changes along with ours. Let us hold until the E/App PR comes up [here](https://github.com/Expensify/App/issues/50174#issuecomment-2427422431).  

Meanwhile can you please upload test videos on all platforms as required by the PR author checklist? Thanks.

**Comment by Skalakid** [comment]
Here is the [PR](https://github.com/Expensify/App/pull/51311) with the Live Markdown library bump, where we will address all the regressions that may appear

**Comment by Skalakid** [comment]
The [PR](https://github.com/Expensify/App/pull/51311) with the Live Markdown bump has been merged

**Comment by rojiphil** [comment]
> The [PR](https://github.com/Expensify/App/pull/51311) with the Live Markdown bump has been merged

Ah! Nice. Thanks @Skalakid. 

@NJ-2020 We can take this off hold. Also, #50990 has bumped the `expensify-common` version to `2.0.100`. That means we only have to bump the version to `2.0.101` so that it matches the one in `parser\dependencies` in `expensify-live-markdown`. 
Can you please merge with the latest main, make the above change, and upload the test videos for all platforms?

**Comment by rojiphil** [comment]
> Okay, will do

@NJ-2020 Can you please give an ETA for this as the issue does not deserve to be on-hold?

**Comment by rojiphil** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/582ff2b3-f475-4a36-824d-9b13ecb8b20b

</details>

<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/51d62215-c8d4-4325-8c26-c4bfd153015b

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/85226bd2-2c17-4cc2-95dc-a2a83541cc00

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/3ac3f88e-c144-4df7-8b0a-25b224cf67af

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/f212e391-5248-4f6b-9dff-67b86f3367e2

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/a4a9b892-5e95-4c5a-a8ad-1de7096b0ef6

</details>

**Review by rojiphil — APPROVED** [review]
@NJ-2020 Thanks. I think being more specific in test cases as shown below can help the QA better:
1. Navigate to any chat e.g. a workspace chat.
2. Enter the message `#4856₹&_#7553` in composer and verify that only `#4856` is highlighted.
3. Send the message and verify that the Concierge message mentions only `#4856` for room creation.
4. Repeat Steps (2) and (3) and verify the expected result:
a. Message: `#gh4h_+&&&-#ehhh`, Expected Result: Only `#gh4h`
b. Message: `#4856₹& #7553`, Expected Result: `#4856`  and `#7553`
c. Message: `#4856₹&`
`#7553`, Expected Result: `#4856`  and `#7553`
d. Message: `#4856₹&*#7553`, Expected Result: Only `#4856`
e. Message: `#4856₹&~#7553`, Expected Result: Only `#4856`
f. Message: `#4856₹& *#7553`, Expected Result: `#4856`  and `#7553`
g. Message: `#4856₹& ~#7553`, Expected Result: `#4856`  and `#7553`
h. Message: `#4856₹&`
`*#7553`, Expected Result: `#4856`  and `#7553`
i. Message: `#4856₹&`
`~#7553`, Expected Result: `#4856`  and `#7553`
j. Message: `#4856₹&`
`_#7553`, Expected Result: `#4856`  and `#7553`
k. Message: `#4856₹&`
`-#7553`, Expected Result: Only `#4856`
l. Message: `#room-name123`, Expected Result: `#room-name123`
m. Message: `https://abc.com/abc#qwer`, Expected Result: Nothing is highlighted

@Beamanator @rlinoz This PR ensures that the changes in [expensify-common PR](https://github.com/Expensify/expensify-common/pull/804) and [react-native-live-markdown PR](https://github.com/Expensify/react-native-live-markdown/pull/521) are working as expected. 
Changes LGTM and works well too. 
Over to you for review. Thanks.

File: package.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
   "version": "9.0.54-1",
   "author": "Expensify, Inc.",
   "homepage": "https://new.expensify.com",
-  "description": "New Expensify is the next generation of Expensify: a reimagination of payments based atop a foundation of chat.",
   "license": "MIT",
+  "description": "New Expensify is the next generation of Expensify: a reimagination of payments based atop a foundation of chat.",
```
Comment (Beamanator) [inline package.json:7]: NAB: Why this change? 🤷

**Review by Beamanator — DISMISSED** [review]

**Review by Beamanator — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (Beamanator) [reply package.json]: thanks for updating!

**Review by rlinoz — APPROVED** [review]
